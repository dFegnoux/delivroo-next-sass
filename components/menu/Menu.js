import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import MenuCategory from "./MenuCategory";
//import PaymentForm from "../forms/PaymentForm";
import Cart from "./Cart";
import "./menu.scss";

class MenuCategories extends PureComponent {
  static propTypes = {
    categories: PropTypes.array,
    updateCart: PropTypes.func.isRequired,
    cart: PropTypes.object.isRequired
  };

  static defaultProps = {
    categories: []
  };

  render() {
    const { categories, updateCart, cart, showPayment } = this.props;

    return (
      <div className="menuLayout">
        <div className="menuContentWrapper">
          {categories.map(category => (
            <MenuCategory
              key={category[0]}
              title={category[0]}
              menus={category[1]}
              updateCart={updateCart}
            />
          ))}
        </div>
        <Cart
          menus={cart}
          updateCart={updateCart}
          disableButtons={showPayment}
        />
      </div>
    );
  }
}

export default MenuCategories;
