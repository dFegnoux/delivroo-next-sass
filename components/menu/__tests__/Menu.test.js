import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Menu from "../Menu";

const mandatoryProps = {
  updateCart: jest.fn(),
  cart: {}
};
const container = shallow(<Menu {...mandatoryProps} />);

describe("Menu snapshots", () => {
  it("should render correctly with default props", () => {
    const tree = renderer.create(<Menu {...mandatoryProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly some categories", () => {
    const categories = Object.entries({
      menu1: [
        {
          title: "mocked title",
          image: "mocked image",
          id: "mocked id",
          description: "mocked description"
        }
      ]
    });
    const tree = renderer
      .create(<Menu categories={categories} {...mandatoryProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
