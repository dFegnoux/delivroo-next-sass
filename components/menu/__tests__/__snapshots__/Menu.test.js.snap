// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshots should render correctly some categories 1`] = `
<div
  className="menuLayout"
>
  <div
    className="menuContentWrapper"
  >
    <div
      className="menuCategory"
    >
      <div
        className="title"
      >
        menu1
      </div>
      <div
        className="itemsList"
      >
        <div
          className="menuCard"
          onClick={[Function]}
        >
          <div
            className="content roundedItem"
          >
            <div
              className="texts"
            >
              <div
                className="cardTitle"
              >
                mocked title
              </div>
              <div
                className="description"
              >
                mocked description
              </div>
              <div
                className="bottom"
              >
                <span
                  className="price"
                >
                   â‚¬
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cartWrapper"
  >
    <div
      className="cart roundedItem"
    >
      <div>
        Votre panier est vide
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu snapshots should render correctly with default props 1`] = `
<div
  className="menuLayout"
>
  <div
    className="menuContentWrapper"
  />
  <div
    className="cartWrapper"
  >
    <div
      className="cart roundedItem"
    >
      <div>
        Votre panier est vide
      </div>
    </div>
  </div>
</div>
`;
